core/controllers/base_test.py

```diff
@@ -269,6 +270,23 @@ def test_i18n_keys(self):
                     self.log_line('- %s' % key)
                 self.log_line('')
 
+    def test_alphabetic_i18n_keys(self):
+        """Tests that the keys of all i18n json files are arranged in
+        alphabetical order."""
+        filenames = os.listdir(
+            os.path.join(os.getcwd(), self.get_static_asset_filepath(),
+                         'assets', 'i18n'))
+        for filename in filenames:
+            with open(os.path.join(os.getcwd(), 'assets', 'i18n', filename),
+                      mode='r') as f:
+                for line in f:
+                    line = line.replace("{", "")
+                    line = line.replace("}", "")
+                    line = line.partition(':')
+                    line = line[0].split()
+                    ord_line = sorted(line)
```

Comment:
This seems to be sorting the line itself (i.e., the tuple containing the key, ':', and the value) rather than checking the sortedness across lines. You should be checking that line is lexicographically after the previous line, and not checking sorting within the line.